src/components/FormValidator.js

```diff
@@ -1,51 +1,59 @@
 export default class FormValidator {
-  constructor(settings, formElement) {
-    this._formElement = formElement;
 
+  constructor(settings, formElement) {
     this._inputSelector = settings.inputSelector;
-    this._submitButton = this._formElement.querySelector(
-      settings.submitButtonSelector
-    );
+    this._submitButtonSelector = settings.submitButtonSelector;
     this._inactiveButtonClass = settings.inactiveButtonClass;
     this._inputErrorClass = settings.inputErrorClass;
     this._errorClass = settings.errorClass;
-
-    this._inputElements = [
-      ...this._formElement.querySelectorAll(this._inputSelector),
-    ];
-  }
-
-  _showInputError(inputElement) {
-    const errorMessageEl = this._formElement.querySelector(
-      `#${inputElement.id}-error`
+    this._form = formElement;
+    this._inputElements = Array.from(
+      formElement.querySelectorAll(this._inputSelector)
     );
-    inputElement.classList.add(this._inputErrorClass);
-    errorMessageEl.textContent = inputElement.validationMessage;
+    this._submitButton = formElement.querySelector(this._submitButtonSelector);
   }
 
-  _hideInputError(inputElement) {
-    const errorMessageEl = this._formElement.querySelector(
-      `#${inputElement.id}-error`
-    );
-    inputElement.classList.remove(this._inputErrorClass);
-    errorMessageEl.textContent = "";
+  _showInputError(inputEl, errorMessageElm) {
+    inputEl.classList.add(this._inputErrorClass);
+    errorMessageElm.textContent = inputEl.validationMessage;
+    errorMessageElm.classList.add(this._errorClass);
   }
 
-  _checkFormValidity() {
-    return this._inputElements.every((input) => input.validity.valid);
+  _hideInputError(formEl, inputEl) {
+    const errorMessageElm = formEl.querySelector(`#${inputEl.id}-error`);
+    inputEl.classList.remove(this._inputErrorClass);
+    errorMessageElm.textContent = " ";
+    errorMessageElm.classList.remove(this._errorClass);
   }
 
-  _checkInputValidity(inputElement) {
-    if (!inputElement.validity.valid) {
-      this._showInputError(inputElement);
+  _checkInputValidity(formEl, inputEl) {
+    if (!inputEl.validity.valid) {
+      this._showInputError(
+        inputEl,
+        formEl.querySelector(`#${inputEl.id}-error`)
+      );
     } else {
-      this._hideInputError(inputElement);
+      this._hideInputError(formEl, inputEl);
     }
   }
 
+  _checkFormValidity = () => {
+    return this._inputElements.every(
+      (input) => input.validity.valid
+    );
+  };
+
+  resetValidation() {
+    this.toggleButtonState();
+
+    this._inputElements.forEach((inputElement) => {
+      this._hideInputError(this._form, inputElement)
+    });
+  }
+
```

Comment:
**Could be improved**

You could make a special method `resetValidation` for clearing errors and controlling the submit button.
It could look something like this:

```js

resetValidation() {
      this._toggleButtonState(); <== controlling the submit button ==

      this._inputList.forEach((inputElement) => {
        this._hideError(inputElement) <== clearing errors ==
      });

    }



```

And you can call it in `index.js` when you click on open buttons